homeassistant/components/bsblan/climate.py

```diff
@@ -0,0 +1,230 @@
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def hvac_mode(self):
+        """Return the current operation mode."""
+        return self._current_hvac_mode
+
+    @property
+    def hvac_modes(self):
+        """Return the list of available operation modes."""
+        return self._hvac_modes
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    @property
+    def preset_modes(self):
+        """List of available preset modes."""
+        return self._preset_modes
+
+    @property
+    def preset_mode(self):
+        """Return the preset_mode."""
+        if self._current_hvac_mode == 2:
+            return PRESET_ECO
+        return PRESET_NONE
+
+    async def async_set_hvac_mode(self, hvac_mode):
+        """Set HVAC mode."""
+        _LOGGER.info("Setting HVAC mode to: %s", hvac_mode)
+        await self.async_set_data(hvac_mode=hvac_mode)
+
+    async def async_set_temperature(self, **kwargs):
+        """Set new target temperatures."""
+        await self.async_set_data(**kwargs)
+
+    async def async_set_data(self, **kwargs: Any) -> None:
+        """Set device settings using BSBLan."""
+        data = {}
+
+        if ATTR_TEMPERATURE in kwargs:
+            data[ATTR_TARGET_TEMPERATURE] = kwargs[ATTR_TEMPERATURE]
+            _LOGGER.info("data = %s", data)
+
+        if ATTR_HVAC_MODE in kwargs:
+            data[ATTR_HVAC_MODE] = HA_STATE_TO_BSBLAN[kwargs[ATTR_HVAC_MODE]]
+            _LOGGER.info("data hvac mode = %s", data)
+
+        try:
+            await self.bsblan.thermostat(**data)
+        except BSBLanError:
+            _LOGGER.error("An error occurred while updating the BSBLan Device")
+            self._available = False
+
+    async def async_update(self) -> None:
+        """Update BSBlan entity."""
+        try:
+            state: State = await self.bsblan.state()
+        except BSBLanError:
+            if self._available:
+                _LOGGER.error("An error occurred while updating the BSBLan device.")
+            self._available = False
+            return
+
+        self._available = True
+
+        self._current_temperature = float(state.current_temperature)
+        self._target_temperature = float(state.target_temperature)
+
+        # check if preset is active
+        if BSBLAN_TO_HA_STATE[state.current_hvac_mode] == 2:
+            self._current_hvac_mode = None
+            self._preset_modes = HA_PRESET_TO_BSBLAN[state.current_hvac_mode]
+        else:
+            self._current_hvac_mode = BSBLAN_TO_HA_STATE[state.current_hvac_mode]
+
+        self._temperature_unit = state.temperature_unit
+
+    @property
+    def device_info(self) -> Dict[str, Any]:
+        """Return device information about this BSBLan device."""
+        return {
+            ATTR_IDENTIFIERS: {(DOMAIN, self._info.device_identification)},
```

Comment:
If `self._info.device_identification` is `None` this id doesn't work.

Reply:
I am a bit confused. This info cannot be None or there would be no device?

Reply:
We're checking for that in the name property. That's why I asked.

Reply:
Bump.